specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,919 @@
+          "description": "If the action type is block, customer can override the response body. The body must be specified in base64 encoding.",
+          "type": "string",
+          "pattern": "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$"
+        }
+      }
+    },
+    "CdnEndpoint": {
+      "description": "Defines the ARM Resource ID for the linked endpoints",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "ARM Resource ID string."
+        }
+      }
+    },
+    "CustomRuleList": {
+      "description": "Defines contents of custom rules",
+      "properties": {
+        "rules": {
+          "description": "List of rules",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomRule"
+          }
+        }
+      }
+    },
+    "CustomRule": {
+      "description": "Defines the common attributes for a custom rule that can be included in a waf policy",
+      "type": "object",
+      "required": [
+        "name",
+        "priority",
+        "matchConditions",
+        "action"
+      ],
+      "properties": {
+        "name": {
+          "description": "Defines the name of the custom rule",
+          "type": "string"
+        },
+        "enabledState": {
+          "description": "Describes if the custom rule is in enabled or disabled state. Defaults to Enabled if not specified.",
+          "type": "string",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "x-ms-enum": {
+            "name": "CustomRuleEnabledState",
+            "modelAsString": true
+          }
+        },
+        "priority": {
+          "description": "Defines in what order this rule be evaluated in the overall list of custom rules",
+          "type": "integer"
+        },
+        "matchConditions": {
+          "description": "List of match conditions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MatchCondition"
+          }
+        },
+        "action": {
+          "description": "Describes what action to be applied when rule matches",
+          "$ref": "#/definitions/ActionType"
+        }
+      }
+    },
+    "RateLimitRuleList": {
+      "description": "Defines contents of rate limit rules",
+      "properties": {
+        "rules": {
+          "description": "List of rules",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RateLimitRule"
+          }
+        }
+      }
+    },
+    "RateLimitRule": {
+      "description": "Defines a rate limiting rule that can be included in a waf policy",
+      "type": "object",
+      "required": [
+        "rateLimitThreshold",
+        "rateLimitDurationInMinutes"
+      ],
+      "properties": {
+        "rateLimitThreshold": {
+          "description": "Defines rate limit threshold.",
+          "type": "integer"
```

Comment:
Does this take negatives? Is there a maximum?

Reply:
No, it cannot take negatives, I added a minimum constraint, thanks for catching this!
There is no maximum in my knowledge at the moment, it will just be the int max for now.